### ISSUE TITLE
Confusing messages with await_only_futures, unnecessary_parenthesis and futures with ternary operator

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#33655]
```yaml
linter:
  rules:
    - await_only_futures
    - unnecessary_parenthesis
```

```dart
import 'dart:async';

main() async {
  bool cond = true;
  await (cond ? Future<void>.value(true) : Future<void>.value(true));
  await cond ? Future<void>.value(true) : Future<void>.value(true);
}
```

I'm sure one of those `await` lines should be valid, but the first one warns `Unnecessary parenthesis can be removed.` and the second one warns `Await only futures.`. I think one of them may be wrong?

**Comment by lrhn** [comment]
The one that is definitely wrong is the unnecessary parentheses warning.
> *awaitExpression* ::= `await` *unaryExpression*

and *conditionalExpression* is not a unary expression.
The analyzer is correct that you can remove the parentheses, it's still syntactically valid, but it won't mean the same thing any more (it'll likely be `(await cond) ? ...` or even awaiting some initial sub-expression of `cond`). In that case you are awaiting a non-future, so that warning is correct.

Ob-nitpick: It should probably say "Unnecessary parentheses can be removed" - removing just one isn't going to work :) Or "Unnecessary parenthesization"

**Comment by srawlins** [comment]
I cannot reproduce a report on the first `await` line. `unnecessary_parenthesis` was fixed perhaps?

**Comment by DanTup** [comment]
I also can't repro on SDK v2.1 either. There have been changed to `unnecessary_parenthesis` since I opened this, so maybe it was one of those.

**Comment by a14n** [comment]
It was fixed by 51b8fc7
